src/check/context/mod.rs

```diff
@@ -183,94 +169,69 @@ impl LookupField<&str, Field> for Context {
     }
 }
 
-#[allow(clippy::large_enum_variant)]
-#[derive(Debug, PartialEq, Eq, Hash)]
+#[derive(Debug, Eq)]
 pub enum ClassVariant {
```

Comment:
ClassVariant and all associated logic could at the very least be in its own file